content/general/concepts/algorithm/terms/euclidean-algorithm/euclidean-algorithm.md

```diff
@@ -0,0 +1,152 @@
+
+```java
+public static void main(String[] args) {
+        System.out.println(gcd(10,15));
+    }
+    public static int gcd(int a,int b){
+        int minValue = Math.min(a,b);
+        while(minValue>0){
+            if(a%minValue==0 && b%minValue==0){
+                break;
+            }
+            minValue--;
+        }
+        return minValue;
+    }
+}
+```
+
+The output for the above code will be:
+
+```shell
+5
+```
+
+## Approach with Euclidean algorithm
+
+Here two methods are shown using Euclidean algorithm to find the HCF of two numbers. So let's take two numbers as 'a' and 'b' as our inputs.
+
+## Method 2
+
+In this method, we aim to compare both 'a' and 'b'. Whichever number is greater, we subtract the smaller number from the larger number and update the value of the larger number accordingly. If 'a' is greater than 'b', we replace 'a' with 'a - b', and if 'b' is greater than 'a', we replace 'b' with 'b - a'. We repeat this step until 'a' becomes equal to 'b'. Once 'a' and 'b' are equal, we return 'a' as the answer. At this point, the value of 'a' (which is equal to 'b') represents the highest common factor (HCF) of the original values of 'a' and 'b'.
+
+```java
+class Euclidean1 {
+static int gcd(int a, int b) {
+
+    while (a != b) {
+        if (a > b)
+            a = a - b;
+        else
+            b = b - a;
+    }
+
+    return a;
+}
+
+public static void main(String[] args) {
+
+    int a = 15, b = 20;
+
+    System.out.println(gcd(a, b));
+
+}
+}
+```
+
+The out for the above code will be:
+
+```shell
+5
+```
+
+If we debug the code above step by steps then:
+
+- Since 'b' is greater than 'a' (20 > 15), we replace 'b' with 'b - a', which gives us 'b = 20 - 15 = 5'.
+- Now 'a' is still 15, and 'b' is 5.
+- We proceed to replace 'a' with 'a - b', resulting in 'a = 15 - 5 = 10'.
+- The values are now 'a = 10' and 'b = 5'.
+- We continue the process and replace 'a' with 'a - b', giving us 'a = 10 - 5 = 5'.
+- Now 'a' and 'b' are both equal to 5.
+- At this point, the while loop exits, and we return 'a' as our answer, which is 5.
+- Therefore, the highest common factor (HCF) of (15, 20) is 5.
+
+## Method 3
+
+In this method we will use [recursion](https://www.codecademy.com/learn/discrete-math-recurrence-relations). The method takes two integer parameters 'a' and 'b' . Then returns an integer as the result. The gcd method uses a `recursive` approach to implement the `Euclidean algorithm`. It first checks if `b` is equal to 0. If it is, then it means that `a` is the GCD, and it returns `a` as the result. If `b` is not 0, it means there is a remainder when `a` is divided by `b`. In this case, the method calls itself recursively with the arguments `b` and `a % b`. This recursive call continues until `b` becomes 0, at which point the base case is triggered, and the GCD is found.
```

Comment:
[Nitpick] See my feedback on [this line](https://github.com/Codecademy/docs/pull/2455#discussion_r1226023651). They share the same problem. 
```suggestion
If we debug the code above step by steps then:

- Since 'b' is greater than 'a' (20 > 15), we replace 'b' with 'b - a', which gives us 'b = 20 - 15 = 5'.
- Now 'a' is still 15, and 'b' is 5.
- We proceed to replace 'a' with 'a - b', resulting in 'a = 15 - 5 = 10'.
- The values are now 'a = 10' and 'b = 5'.
- We continue the process and replace 'a' with 'a - b', giving us 'a = 10 - 5 = 5'.
- Now 'a' and 'b' are both equal to 5.
- At this point, the while loop exits, and we return 'a' as our answer, which is 5.
- Therefore, the highest common factor (HCF) of (15, 20) is 5.

## Method 3

In this method we will use [recursion](https://www.codecademy.com/learn/discrete-math-recurrence-relations). The method takes two integer parameters 'a' and 'b' . Then returns an integer as the result. The gcd method uses a `recursive` approach to implement the `Euclidean algorithm`. It first checks if `b` is equal to 0. If it is, then it means that `a` is the GCD, and it returns `a` as the result. If `b` is not 0, it means there is a remainder when `a` is divided by `b`. In this case, the method calls itself recursively with the arguments `b` and `a % b`. This recursive call continues until `b` becomes 0, at which point the base case is triggered, and the GCD is found.
```